### ISSUE TITLE
Possible source of error in creating master connection string

### ISSUE THREAD

**Issue by halcwb** [Zaid-Ajaj/ThrowawayDb#13]
Hello @Zaid-Ajaj.  I am rewriting your code to the F# equivalent and I stumbled upon a problem with this line of code:

```csharp
var connectionStringOfMaster = ConnectionString.Replace(databaseName, "master");
```

By change I tested this with a database name that was part of the Data Source name. I,e.

```fsharp
let connString = @"Data Source=VOXDB-PICURED01;Initial Catalog=PICURED;Persist Security Info=True;Integrated Security=SSPI;"
```
And obviously replacing PICURED with master yields a Data Source=VOXDB-master01

**Comment by Zaid-Ajaj** [comment]
Hi Casper, I see what is happening here. The thing is, we are not using just a simple string replace. Instead, we use `SqlConnectionStringBuilder` to parse the full database name (i.e. "VOXDB-PICURED01" in your case) and replace it by "master". I am not sure why in your case, only "PICURED" would get replaced and not the full database name "VOXDB-PICURED01" 

On a side note: why rewrite the library in F# if you can use this library underneath and only wrap it inside a more F# API :wink:? Personally I have used library in F# projects without problems but it is up to if you want a different API 😄

**Comment by halcwb** [comment]
@Zaid-Ajaj Yep you're right, I did overlook that. And I would use you're lib. Only I want it to run on the new Microsoft.Data.SqlClient. Also, I use it as a sort of an exercise project.

**Comment by Zaid-Ajaj** [comment]
> Only I want it to run on the new Microsoft.Data.SqlClient.

Sounds like it would be a nice and easy PR to add to this library 😉

**Comment by halcwb** [comment]
> Hi Casper, I see what is happening here. The thing is, we are not using just a simple string replace. Instead, we use `SqlConnectionStringBuilder` to parse the full database name (i.e. "VOXDB-PICURED01" in your case) and replace it by "master". I am not sure why in your case, only "PICURED" would get replaced and not the full database name "VOXDB-PICURED01"
> 
> On a side note: why rewrite the library in F# if you can use this library underneath and only wrap it inside a more F# API ? Personally I have used library in F# projects without problems but it is up to if you want a different API

I had a closer look at you'r code, but I think it is a simple string replace, only this method is used with the generated connection string that contains a random generated database name like: Data Source=VOXDB-PICURED01;Initial Catalog=ThrowawayDbfe50108b74;Integrated Security=True;Persist Security Info=True

So this will very unlikely be confused with the Data Source name. 

> > Only I want it to run on the new Microsoft.Data.SqlClient.
> 
> Sounds like it would be a nice and easy PR to add to this library

I might try that, but don't underestimate how rusty my C# language abilities are ;-).

**Comment by Zaid-Ajaj** [comment]
I see what you mean, yeah then this still remains. I'll reopen and try to fix it soon :smile:

**Comment by halcwb** [comment]
@Zaid-Ajaj And while I am at it, this one is a bit silly but still:

```fsharp
#r "nuget: ThrowAwayDb"

open ThrowawayDb

let connString = 
    @"Data Source=foobar;Initial Catalog=foobar;Persist Security Info=True;Integrated Security=SSPI;Connection Timeout=0;"

do
    use db = ThrowawayDatabase.Create(connString)
    printfn $"connection string: {db.ConnectionString}"

```

Well to be true, I did really ran into this issue as I used a connection string for long running db ops. The problem is that the TryPingDatabase method doesn't restrict the connection timeout.

**Comment by Zaid-Ajaj** [comment]
I believe it uses the default timeout of 30 seconds which I admit can be annoying when you have multiple tests that all fail. Maybe I can add it an another option to the `Create` function:
```fs
use db = ThrowawayDatabase.Create(connString, timeout=30)
```

**Comment by halcwb** [comment]
If you do something like this:

```fsharp
[<RequireQualifiedAccess>]
module SqlConnectionStringBuilder =

    open Microsoft.Data.SqlClient

    let tryCreate connString =
        try
            SqlConnectionStringBuilder(connString)
            |> Some
        with
        | e ->
            printfn $"cannot create the connection string builder:\n{e.Message}"
            None
```

You'll have a builder and then you can set anything you like for the connection string like:

```fsharp
SqlConnectionStringBuilder.tryCreate ""
|> function
| Some builder ->
    builder.InitialCatalog <- "master"
    builder.CommandTimeout <- 1
    builder.ConnectTimeout <- 1
    builder.ConnectionString
| None ->  
    ""
```

I think this is safer then a string replace on the connection string.

**Comment by Zaid-Ajaj** [comment]
Thanks a lot Casper, I will give it a try soon ✔️

**Comment by Zaid-Ajaj** [comment]
Hi @halcwb I just published a new version of `ThrowawayDb`  v1.5 where it fixes this problem with the connection string replacement and uses latest `Microsoft.Data.SqlClient` instead of `System.Data.SqlClient` please feel free to give it a try if you and let me know how it goes for you 😄

**Comment by halcwb** [comment]
@Zaid-Ajaj Looks great, thanks. FYI this is my F# translation: https://gist.github.com/halcwb/c7a1340d69c9d4dcd2c6f0cdda46ab7e.

Small detail, I don't think you need to keep track of the original connection string. Just use the master connection string for that.
